lms/services/email_preferences.py

```diff
@@ -78,11 +113,10 @@ def _url(self, route, h_userid, tag):
             _query={"token": self._encode_token(TokenPayload(h_userid, tag))},
         )
 
-    def unsubscribe(self, h_userid):
+    def instructor_digest_unsubscribe(self, h_userid) -> None:
         """Unsubscribe `h_userid` from emails of type `tag`."""
-        self.set_preferences(
-            EmailPrefs(h_userid, **{day: False for day in EmailPrefs.DAYS})
-        )
+        email_preferences = self.get_preferences(h_userid)
```

Comment:
We first get the current preferences and the unsubscribe from digest.

This is superfluous now because digest is the only preference but it will be important once we have more types, we don't want to just ovrride every value.